src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -165,30 +165,31 @@ public HttpResponse doProceed(StaplerRequest request) throws IOException, Servle
      * Processes the acceptance (approval) request.
      * This method is used by both {@link #doProceedEmpty()} and {@link #doProceed(StaplerRequest)}
      *
-     * @param params A map that represents the parameters sent in the request
+     * @param v The parameters sent in the request
      * @return A HttpResponse object that represents Status code (200) indicating the request succeeded normally.
      */
-    public HttpResponse proceed(@CheckForNull Map<String,Object> params) {
+    @SuppressWarnings("unchecked")
+    public HttpResponse proceed(@CheckForNull Object v) {
         User user = User.current();
         String approverId = null;
+        Object returnObject = v;
         if (user != null){
             approverId = user.getId();
             run.addAction(new ApproverAction(approverId));
             listener.getLogger().println("Approved by " + hudson.console.ModelHyperlinkNote.encodeTo(user));
         }
-        node.addAction(new InputSubmittedAction(approverId, params));
-
-        Object v;
-        if (params != null && params.size() == 1) {
-            v = params.values().iterator().next();
-        } else {
-            v = params;
+        if (v != null && v instanceof Map) {
```

Comment:
This is not a good idea.

Revert. Have the `Map` overload be the real implementation. Restore the original method overload with `Object`, deprecate it, and have it delegate to the new overload.